File: session-3/page-objects-checkout/python/pages/cart_page.py
"""CartPage — the "My Cart" screen. Confirms the cart loaded and starts checkout.

Locators carried over verified from the Session 2 search->checkout capstone.
"""

from appium.webdriver.common.appiumby import AppiumBy
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC


class CartPage:
    # --- locators ---
    TITLE = (AppiumBy.ACCESSIBILITY_ID, "My Cart")
    PROCEED_TO_CHECKOUT = (AppiumBy.ANDROID_UIAUTOMATOR, 'new UiSelector().className("android.widget.Button").description("Proceed to Checkout")')

    def __init__(self, driver):
        self.driver = driver

    def wait_loaded(self):
        """Wait until the cart screen is shown."""
        WebDriverWait(self.driver, 20).until(EC.visibility_of_element_located(self.TITLE))

    def checkout(self):
        """Confirm the cart loaded, then proceed to checkout."""
        self.wait_loaded()
        WebDriverWait(self.driver, 20).until(EC.element_to_be_clickable(self.PROCEED_TO_CHECKOUT)).click()

File: session-3/page-objects-checkout/node/test/pages/cart.page.js
// CartPage — the "My Cart" screen. Confirms the cart loaded and starts checkout.
//
// Locators carried over verified from the Session 2 search->checkout capstone.

class CartPage {
  // --- locators ---
  get title() {
    return $('~My Cart')
  }
  get proceedToCheckoutButton() {
    return $('android=new UiSelector().className("android.widget.Button").description("Proceed to Checkout")')
  }

  /** Wait until the cart screen is shown. */
  async waitLoaded() {
    await this.title.waitForDisplayed({ timeout: 20000 })
  }

  /** Confirm the cart loaded, then proceed to checkout. */
  async checkout() {
    await this.waitLoaded()
    await this.proceedToCheckoutButton.click()
  }
}

export default new CartPage()
